ragna/ui/central_view.py

```diff
@@ -0,0 +1,681 @@
+            </svg>
+            <span>${title}</span>
+        </div>            
+        """
+
+    _scripts = {
+        "copy_to_clipboard": """navigator.clipboard.writeText(`${data.value}`);"""
+    }
+
+    _stylesheets = [
+        """ div.container { 
+                            
+                            display:flex;
+                            flex-direction: row;
+                            margin: 7px 10px;
+                    
+                            color:gray;
+                        } 
+                    
+                        svg {
+                            margin-right:5px;
+                        }
+
+                    """
+    ]
+
+
+class RagnaChatMessage(pn.chat.ChatMessage):
+    msg_data = param.Dict(default={})
+    on_click_source_info_callback = param.Callable(default=None)
+
+    def __init__(self, msg_data, user, on_click_source_info_callback=None):
+        self.role = msg_data["role"]
+
+        params = {
+            "msg_data": msg_data,
+            # user is the name of the assistant (eg 'Ragna/DemoAssistant')
+            # or the name of the user, depending on the role
+            "user": user,
+            "on_click_source_info_callback": on_click_source_info_callback,
+            "object": msg_data["content"],
+            "renderers": [
+                lambda txt: RagnaChatMessage.chat_entry_value_renderer(
+                    txt, role=self.role
+                )
+            ],
+            "show_timestamp": False,
+            "show_reaction_icons": False,
+            "show_copy_icon": False,
+            "show_user": False,
+        }
+
+        params["avatar"] = RagnaChatMessage.get_avatar(self.role, user)
+
+        super().__init__(**params)
+
+        self.update_css_classes()
+        self.chat_copy_icon.visible = False
+
+        if self.role != "user":
+            source_info_button = pn.widgets.Button(
+                name="Source Info",
+                icon="info-circle",
+                stylesheets=[
+                    ui.CHAT_INTERFACE_CUSTOM_BUTTON,
+                ],
+            )
+
+            source_info_button.on_click(self.trigger_on_click_source_info_callback)
+
+            copy_button = RagnaChatCopyIcon(
+                value=self.object,
+                title="Copy",
+                stylesheets=[
+                    ui.CHAT_INTERFACE_CUSTOM_BUTTON,
+                ],
+            )
+
+            self._composite[1].append(pn.Row(copy_button, source_info_button, height=0))
+
+    def trigger_on_click_source_info_callback(self, event):
+        if self.on_click_source_info_callback is not None:
+            self.on_click_source_info_callback(event, self)
+
+    def update_css_classes(self):
+        role = self.msg_data["role"] if "role" in self.msg_data else None
+        self.css_classes = [
+            "chat-entry",
+            "chat-entry-user" if role == "user" else "chat-entry-ragna",
```

Comment:
```suggestion
            f"chat-entry-{role}",
```

Reply:
Well that's not that simple. This would lead to have a CSS class `chat-entry-system`, which doesn't exist - unless I define it as identical to `chat-entry-ragna` which is supposed to be the default... I'll sort something out.